shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m

```diff
@@ -896,6 +899,35 @@ - (void)testGarbageInputViewsAreNotRemovedImmediately {
   [self commitAutofillContextAndVerify];
 }
 
+- (void)testDecommissionedViewAreNotReusedByAutofill {
+  // Regression test for https://github.com/flutter/flutter/issues/84407.
+  NSMutableDictionary* configuration = self.mutableTemplateCopy;
+  [configuration setValue:@{
+    @"uniqueIdentifier" : @"field1",
+    @"hints" : @[ UITextContentTypePassword ],
+    @"editingValue" : @{@"text" : @""}
+  }
+                   forKey:@"autofill"];
+  [configuration setValue:@[ [configuration copy] ] forKey:@"fields"];
+
+  [self setClientId:123 configuration:configuration];
+
+  [self setTextInputHide];
+  UIView* previousActiveView = textInputPlugin.activeView;
+
+  [self setClientId:124 configuration:configuration];
+
+  // Make sure the autofillable view is reused.
+  XCTAssertEqual(previousActiveView, textInputPlugin.activeView);
+  XCTAssertNotNil(previousActiveView);
+  // Does not crash.
+}
+
+- (void)testInitialActiveViewCantAccessTextInputDelegate {
```

Comment:
Can you add a comment here since reading this test is very confusing and not clear what we are asserting.